src/client.ts

```diff
@@ -4097,7 +4098,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         if (!eventId?.startsWith(EVENT_ID_PREFIX)) {
             opts = txnId as IRedactOpts;
             txnId = eventId;
-            eventId = threadId;
+            eventId = threadId!;
```

Comment:
Likewise, why does it make sense to say this is guaranteed non-null?

Reply:
Due to the overload signatures, if eventId doesn't start with event id prefix then we know which overload we are in and can make more deductions than typescript is able to do for us